tests/Microsoft.Identity.Test.Unit/ManagedIdentityTests/ImdsTests.cs

```diff
@@ -14,180 +14,28 @@ namespace Microsoft.Identity.Test.Unit.ManagedIdentityTests
-                        .WithHttpManager(httpManager)
-                        .WithExperimentalFeatures()
-                        .Build();
-
-                httpManager.AddManagedIdentityMockHandler(ImdsEndpoint, "scope", "", ManagedIdentitySourceType.IMDS, statusCode: HttpStatusCode.InternalServerError);
-                httpManager.AddManagedIdentityMockHandler(ImdsEndpoint, "scope", "", ManagedIdentitySourceType.IMDS, statusCode: HttpStatusCode.InternalServerError);
-
-                MsalServiceException ex = await Assert.ThrowsExceptionAsync<MsalServiceException>(async () =>
-                    await cca.AcquireTokenForClient(new string[] { "scope" })
-                    .WithManagedIdentity()
-                    .ExecuteAsync().ConfigureAwait(false)).ConfigureAwait(false);
-
-                Assert.IsNotNull(ex);
-                Assert.AreEqual(MsalError.ManagedIdentityRequestFailed, ex.ErrorCode);
-                Assert.AreEqual(MsalErrorMessage.ManagedIdentityNoResponseReceived, ex.Message);
-            }
-        }
-
-        [TestMethod]
-        public async Task ImdsNullResponseAsync()
-        {
-            using (new EnvVariableContext())
-            using (var httpManager = new MockHttpManager())
-            {
-                SetEnvironmentVariables("http://169.254.169.254");
-
-                IConfidentialClientApplication cca = ConfidentialClientApplicationBuilder
-                        .Create("clientId")
-                        .WithHttpManager(httpManager)
-                        .WithExperimentalFeatures()
-                        .Build();
-
-                httpManager.AddManagedIdentityMockHandler(ImdsEndpoint, "https://management.azure.com", "", ManagedIdentitySourceType.IMDS, statusCode: HttpStatusCode.OK);
-
-                MsalServiceException ex = await Assert.ThrowsExceptionAsync<MsalServiceException>(async () =>
-                    await cca.AcquireTokenForClient(new string[] { "https://management.azure.com" })
-                    .WithManagedIdentity()
-                    .ExecuteAsync().ConfigureAwait(false)).ConfigureAwait(false);
-
-                Assert.IsNotNull(ex);
-                Assert.AreEqual(MsalError.ManagedIdentityRequestFailed, ex.ErrorCode);
-                Assert.AreEqual(MsalErrorMessage.ManagedIdentityInvalidResponse, ex.Message);
-            }
-        }
-
         [TestMethod]
         public async Task ImdsBadRequestTestAsync()
         {
             using (new EnvVariableContext())
             using (var httpManager = new MockHttpManager())
 
             {
-                SetEnvironmentVariables("http://169.254.169.254");
+                ManagedIdentityTests.SetEnvironmentVariables(ManagedIdentitySourceType.IMDS, "http://169.254.169.254");
 
-                IConfidentialClientApplication cca = ConfidentialClientApplicationBuilder
-                    .Create("clientId")
-                    .WithHttpManager(httpManager)
```

Comment:
I think you should have some tests (separate from tests that focus on source) about each `With*` method you exposed here. Nothing crazy, but just to make sure they are respected.

Reply:
I have added some tests for the builder [here](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/nebharg%2FMsiApiPrototype/tests/Microsoft.Identity.Test.Unit/AppConfigTests/ManagedIdentityApplicationBuilderTests.cs#L29)